File: fold an array.cc
#include <stdlib.h>
#include <stdio.h>
#include <string>
#include <vector>

/*
credits to https://www.codewars.com/users/user5036852

https://www.codewars.com/kata/57ea70aa5500adfe8a000110
*/

std::vector<int> foldArray(std::vector<int> array, int runs){

    std::vector<int> newarray = array;
    for (int j = 0; j < runs; j++){
        std::vector<int> temp;
        for (int i = 0; i < newarray.size() / 2; i++){
            int one = newarray[i];
            int two = newarray[newarray.size() - 1];
            temp.push_back(newarray[i] + newarray[newarray.size() - 1 - i]);
        }
        if (newarray.size() % 2 == 1){
            temp.push_back(newarray[newarray.size() / 2]);
        }
        newarray.clear();
        for (int i = 0; i < temp.size(); i++){
            newarray.push_back(temp[i]);
        }
    }

    return newarray;

}

File: fold_an_array.py
def foldArray(array, runs):
    
    newarray = []
    for integer in array:
        newarray.append(integer)

    for i in range(runs):
        temp = []
        for j in range(int(len(newarray) / 2)):
            temp.append(newarray[j] + newarray[len(newarray) - 1 - j])
        if len(newarray) % 2 == 1:
            temp.append(newarray[int(len(newarray) / 2)])
        newarray.clear()
        for integer in temp:
            newarray.append(integer)

    return newarray
